src/Service.Tests/Unittests/ConfigValidationUnitTests.cs

```diff
@@ -2099,6 +2099,72 @@ string[] linkingTargetFields
             Assert.AreEqual(finalConfigFilePath, runtimeConfigLoader.ConfigFilePath);
         }
 
+        /// <summary>
+        /// Method to validate that runtimeConfig is successfully set up using constructor
+        /// where members are passed in without json config file.
+        /// RuntimeConfig has two constructors, one that loads from the config json and one that takes in all the members.
+        /// This test makes sure that the constructor that takes in all the members works as expected.
+        /// </summary>
+        [TestMethod]
+        public void TestRuntimeConfigSetupWithNonJsonConstructor()
+        {
+            EntitySource entitySource = new(
+                    Type: EntitySourceType.Table,
+                    Object: "sourceName",
+                    Parameters: null,
+                    KeyFields: null
+                );
+
+            Entity sampleEntity1 = new(
+                Source: entitySource,
+                GraphQL: null,
+                Rest: null,
+                Permissions: null,
+                Mappings: null,
+                Relationships: null);
+
+            string entityName = "SampleEntity1";
+            string dataSourceName = "Test1";
+
+            Dictionary<string, Entity> entityMap = new()
+            {
+                { entityName, sampleEntity1 }
+            };
+
+            DataSource ds = new(DatabaseType: DatabaseType.MSSQL, "", Options: null);
+            Dictionary<string, DataSource> dataSourceNameToDataSource = new()
+            {
+                { dataSourceName, ds }
+            };
+
+            Dictionary<string, string> entityNameToDataSourceName = new()
+            {
+                { entityName, dataSourceName }
+            };
+
+            RuntimeConfig runtimeConfig = new(
+                Schema: "UnitTestSchema",
+                DataSource: ds,
+                Runtime: new(
+                    Rest: new(),
+                    GraphQL: new(),
+                    Host: new(null, null)
+                ),
+                Entities: new RuntimeEntities(entityMap),
+                DefaultDataSourceName: dataSourceName,
+                DataSourceNameToDataSource: dataSourceNameToDataSource,
+                EntityNameToDataSourceName: entityNameToDataSourceName
+            );
+
+            Assert.AreEqual(ds, runtimeConfig.DataSource, "RuntimeConfig datasource must match datasource passed into constructor");
```

Comment:
can you have a more descriptive name than `ds`

Reply:
fixed.